### PR TITLE
Make `-library-level` a driver argument and pass it to the frontend

### PR DESCRIPTION
**Pull request by xymus**
rdar://82458987

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please test

**Comment by xymus** [comment]
@artemcm Did I forget something on how to pass down a flag? I'm not aware of the good practices here.

File: Sources/SwiftOptions/Options.swift [hunk 1]
```diff
@@ -377,7 +377,7 @@ extension Option {
   public static let j: Option = Option("-j", .joinedOrSeparate, attributes: [.doesNotAffectIncrementalBuild], metaVar: "<n>", helpText: "Number of commands to execute in parallel")
   public static let LEQ: Option = Option("-L=", .joined, alias: Option.L, attributes: [.frontend, .doesNotAffectIncrementalBuild, .argumentIsPath], group: .linkerOption)
   public static let libc: Option = Option("-libc", .separate, helpText: "libc runtime library to use")
-  public static let libraryLevel: Option = Option("-library-level", .separate, attributes: [.helpHidden, .frontend, .noDriver], metaVar: "<level>", helpText: "Library distribution level 'api', 'spi' or 'other' (the default)")
+  public static let libraryLevel: Option = Option("-library-level", .separate, attributes: [.helpHidden, .frontend, .moduleInterface], metaVar: "<level>", helpText: "Library distribution level 'api', 'spi' or 'other' (the default)")
```
Comment (cltnschlosser) [inline Sources/SwiftOptions/Options.swift:380]: This file is auto generated, did you make this change manually?

**Review by cltnschlosser — COMMENTED** [review]
Yep, this is best way to forward an argument.

File: Sources/SwiftOptions/Options.swift (hunk 1 above)
Comment (xymus) [reply Sources/SwiftOptions/Options.swift:380]: No, I generated it from a yet-to-land similar change on the compiler side. I think I have to merge this one first as the compiler change will end up testing this driver change.

**Review by artemcm — APPROVED** [review]

**Pull request merged**